internal/config/config_test.go

```diff
@@ -0,0 +1,74 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestNewConfig(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   map[string]string
+		want    Config
+		wantErr bool
+	}{
+		{
+			name: "OK",
+			input: map[string]string{
+				"ADDRESS":          "address",
+				"PORT":             "port",
+				"JWT_TOKEN_SECRET": "secret",
+			},
+			want: Config{
+				Address:        "address",
+				Port:           "port",
+				JWTTokenSecret: "secret",
+			},
+		},
+		{
+			name: "empty port",
+			input: map[string]string{
+				"ADDRESS":          "address",
+				"JWT_TOKEN_SECRET": "secret",
+				"PORT":             "",
+			},
+			wantErr: true,
+		},
+		{
+			name: "empty address",
+			input: map[string]string{
+				"PORT":    "port",
+				"":        "secret",
```

Comment:
Here you missing both secret (as it has empty name) and address, as it is set to empty string.
Also, you don't need to set empty environment, because unset env variable will return empty string by default. So you should set only those envs that you need to be set.

Reply:
didn't know about `t.Setenv`,  `os.Setenv` was setting env vars for whole process, that's why i was using empty values to reset them (empty keys my bad, didn't notice), now fixed, thank you

Reply:
seems fixed